pyasic/miners/backends/luxminer.py

```diff
@@ -146,6 +147,23 @@ async def reboot(self) -> bool:
     async def get_config(self) -> MinerConfig:
         return self.config
 
+    async def upgrade_firmware(self) -> bool:
+        """
+        Upgrade the firmware on a LuxOS miner by calling the 'updaterun' API command.
+
+        Returns:
+            bool: True if the firmware upgrade was successfully initiated, False otherwise.
+        """
+        try:
+            await self.rpc.upgraderun()
+
+            logging.info(f"{self.ip}: Firmware upgrade initiated successfully.")
+            return True
+
+        except APIError as e:
+            logging.error(f"{self.ip}: Firmware upgrade failed: {e}")
+            return False
+
```

Comment:
Need to return a status here too, or you can drop the `return False` down one level.

Reply:
Updated, now `upgrade_firmware` to `return tuple`with status and message.

Reply:
If you're going to make it a tuple, it will have to be changed for all other update_firmware functions as well, so might as well just leave it as bool return type.